src/ng/directive/ngTransclude.js

```diff
@@ -163,41 +163,56 @@ var ngTranscludeDirective = ['$compile', function($compile) {
+
+        if (!$transclude) {
+          throw ngTranscludeMinErr('orphan',
+          'Illegal use of ngTransclude directive in the template! ' +
+          'No parent directive that requires a transclusion found. ' +
+          'Element: {0}',
+          startingTag($element));
+        }
 
-      function ngTranscludeCloneAttachFn(clone, transcludedScope) {
-        if (clone.length) {
-          $element.empty();
-          $element.append(clone);
-        } else {
-          // Since this is the fallback content rather than the transcluded content,
-          // we compile against the scope we were linked against rather than the transcluded
-          // scope since this is the directive's own content
-          $compile($element.contents())($scope);
 
-          // There is nothing linked against the transcluded scope since no content was available,
-          // so it should be safe to clean up the generated scope.
-          transcludedScope.$destroy();
+        // If the attribute is of the form: `ng-transclude="ng-transclude"` then treat it like the default
+        if ($attrs.ngTransclude === $attrs.$attr.ngTransclude) {
+          $attrs.ngTransclude = '';
         }
-      }
+        var slotName = $attrs.ngTransclude || $attrs.ngTranscludeSlot;
 
-      if (!$transclude) {
-        throw ngTranscludeMinErr('orphan',
-        'Illegal use of ngTransclude directive in the template! ' +
-        'No parent directive that requires a transclusion found. ' +
-        'Element: {0}',
-        startingTag($element));
-      }
+        // If the slot is required and no transclusion content is provided then this call will throw an error
+        $transclude(ngTranscludeCloneAttachFn, null, slotName);
 
-      // If there is no slot name defined or the slot name is not optional
-      // then transclude the slot
-      var slotName = $attrs.ngTransclude || $attrs.ngTranscludeSlot;
-      $transclude(ngTranscludeCloneAttachFn, null, slotName);
+        // If the slot is optional and no transclusion content is provided then use the fallback content
+        if (slotName && !$transclude.isSlotFilled(slotName)) {
+          useFallbackContent();
+        }
+
+        function ngTranscludeCloneAttachFn(clone, transcludedScope) {
+          if (clone.length) {
+            $element.append(clone);
+          } else {
+            useFallbackContent();
+            // There is nothing linked against the transcluded scope since no content was available,
+            // so it should be safe to clean up the generated scope.
+            transcludedScope.$destroy();
+          }
+        }
+
+        function useFallbackContent() {
+          // Since this is the fallback content rather than the transcluded content,
+          // we link against the scope of this directive rather than the transcluded scope
+          fallbackLinkFn($scope, function(clone) {
+            $element.append(clone);
```

Comment:
Doesn't this mean, that if the fallback content requires a parent controller, it won't be able to find it (since we append it to the DOM after it has been linked)?